### ISSUE TITLE
Carousel state not updating if child is added to items.

### ISSUE THREAD

**Issue by NinjaInShade** [sag1v/react-elastic-carousel#163]
**Describe the bug**
I have the a carousel with some div's being mapped out in React. These div's are in a state, and when i added a new item to that state, the carousel doesn't update. The element is placed in as i've inspected in dev tools, but the spacing isn't being updated and still acts like there were the same amount of elements as before. If there is nothing in state to start of with it works, however a different issue comes up where the items don't stretch to width: 100% like is said in styling.

**To Reproduce**
1. Setup a carousel. I used only the breakpoints prop, and this was it: 
```js
  const breakPoints = [
    { width: 1, itemsToShow: 1 },
    { width: 750, itemsToShow: 2 },
    { width: 1000, itemsToShow: 3 },
    { width: 1500, itemsToShow: 4 },
    { width: 2000, itemsToShow: 5 },
  ];
 ```
 2) Make some state with some data. Include a way to add to the state
3) Map out this state in the carousel as a div. The div's styling ( i removed some styling like transitions that are irrelevant ) i had was:
```css
  border-radius: 8px;
  padding: 30px;
  margin: 8px;
  width: 100%;
```
4) Ensure you have atleast one item in state and add to the state.
5) The new item should have not been visually rendered but can be seen in dev tools. That's the bug.

**Expected behavior**
I expect the items to shift width to allow for the next element to appear on the same row (if breakpoints allow for it) or atleast let me use the arrows to show it.

**Screenshots**
2 state items are being rendered fine
<https://prnt.sc/10m7utv>
New state is about to be added
<https://prnt.sc/10m7vrq>
Been added, it is in the DOM but carousel isn't updating
<https://prnt.sc/10m7wwe>
On refresh of the screen it fixes itself ( my data is from db so it is persistent ) 
https://prnt.sc/10m7xuk

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser chrome
 - Version chrome 89

**Comment by sag1v** [comment]
Can you share how you update that state? I tried it and it works fine

**Comment by NinjaInShade** [comment]
> Can you share how you update that state? I tried it and it works fine

```js
   <Carousel breakPoints={breakPoints}>
              {currentCollection.flashcards.map((flashcard, index) => {
                return (
                  <Flashcard
                    frontContent={flashcard.question}
                    backContent={flashcard.answer}
                    key={index}
                    id={flashcard._id}
                    currentCollectionId={collectionId}
                  />
                );
              })}
            </Carousel>
            ```
            Then currentCollection is just a local state to that component, which is set with this useEffect :
            ```js
             useEffect(() => {
    setCurrentCollection(auth.collections.filter((collection) => collection._id === collectionId)[0]);
  }, [auth.collections, collectionId]);
  ```
  
  So on the component load, it finds the exact collection based on a collectionId prop that is passed down. The collection's are saved in a global Context

**Comment by NinjaInShade** [comment]
This is another instance of where i use the carousel, which directly loads the items from the auth state:

```js
<Carousel breakPoints={breakPoints} key={auth.collections}>
              {auth.collections.map((collection, index) => {
                return (
                  <CollectionCard
                    id={collection._id}
                    key={index}
                    name={collection.name}
                    icon={icons[collection.icon]}
                    flashcardAmount={collection.flashcards.length}
                  />
                );
              })}
</Carousel>
```

**Comment by sag1v** [comment]
Thanks!
Well, I tried it and it seems fine

https://user-images.githubusercontent.com/9153149/111158799-2318f880-85a1-11eb-9f7d-97fd68696485.mov

---

You can try it yourself
https://codesandbox.io/s/react-elastic-carousel-forked-718w4?file=/src/index.js

**Comment by NinjaInShade** [comment]
> Thanks!
> Well, I tried it and it seems fine
> 
>  Screen.Recording.2021-03-15.at.15.12.28.mov 
> You can try it yourself
> https://codesandbox.io/s/react-elastic-carousel-forked-718w4?file=/src/index.js

It must be with my code then. I'll look at it later, but it might have to do with the items being rendered are from a card flip library. I'll get back with what i find.

**Comment by luvpack** [comment]
Same trouble

**Comment by NinjaInShade** [comment]
> Same trouble. Child was added but width of his dont change

I didn't manage to get it fixed but it's on my list still. Did you item width change when you resized the screen in devtools or refreshed the page? That's what was happening to me.

**Comment by luvpack** [comment]
> > Same trouble. Child was added but width of his dont change
> 
> I didn't manage to get it fixed but it's on my list still. Did you item width change when you resized the screen in devtools or refreshed the page? That's what was happening to me.

Yes, after refreshing the page everything is ok

**Comment by sag1v** [comment]
Make sure you are not mutating the state, react won't re-render if the state gets mutated, the Carousel won't "know" that the children were changed because they got the same reference.

**Comment by eacanalg** [comment]
@sag1v Hey got a little update on this. The carrousel has got problems when starting from an empty array (having no children). In you code change useState([1]) to useState([]). Then when scrolling to second window you will keep seeing the first one.

**Comment by sag1v** [comment]
@eacanalg Good finding! resolved as of version `0.11.3`
